### PR TITLE
fix(dedicated.helpcenter): add modal for helpcenter for ca and eu

### PR DESCRIPTION
**Pull request by stif59100**
fix(dedicated.helpcenter): add modal for helpcenter for ca and eu

ref:MANAGER-10138

| Question         | Answer
| ---------------- | ---
 Branch?          | `release/prod-2022-w47` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-10138
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

I created a modal for new helpcenter on the support manager ticket.

## Related

:flags: Translation
- [x] TRANS-47039

### CODE REVIEW

File: packages/manager/modules/support/src/modal/modal.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            <oui-button
+                on-click="$ctrl.goBack()"
+                variant="link"
+                variant-nav="next"
+            >
+                <span data-translate="modal_beta_help_center_later"></span>
+            </oui-button>
+        </div>
+        <div class="col-6">
+            <img data-ng-src="{{ $ctrl.illustration }}" alt="" />
```
Comment (vks-ovh) [inline packages/manager/modules/support/src/modal/modal.html:23]: Can we put some good name for `alt` instead of keeping it empty ?

**Review by vks-ovh — APPROVED** [review]

File: packages/manager/modules/support/src/support.component.js [hunk 2]
```diff
@@ -5,4 +5,5 @@ export default {
   controller,
   name: 'support',
   template,
+  bindings: { goToHelpCenterModal: '<' },
```
Comment (ghyenne) [inline packages/manager/modules/support/src/support.component.js:8]: No need "goToHelpCenterModal: '<'"

File: packages/manager/modules/support/src/modal/modal.constants.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+export const EVENT_NAMESPACE = 'modal.events';
+
+export const FEATURES = ['support:betaModalAccess'];
+
+export const EVENT_NAMES = {
```
Comment (ghyenne) [inline packages/manager/modules/support/src/modal/modal.constants.js:5]: isn't use.

File: packages/manager/modules/support/src/modal/modal.constants.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+export const EVENT_NAMESPACE = 'modal.events';
```
Comment (ghyenne) [inline packages/manager/modules/support/src/modal/modal.constants.js:1]: Isn't use

File: packages/manager/modules/support/src/modal/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  /* @ngInject */ ($stateProvider) => {
+    $stateProvider.state('support.tickets.beta-help-center.**', {
+      lazyLoad: ($transition$) => {
+        const $ocLazyLoad = $transition$.injector().get('$ocLazyLoad');
+
+        return import('./modal.module').then((mod) =>
+          $ocLazyLoad.inject(mod.default || mod),
+        );
+      },
+      url: '/modal',
```
Comment (ghyenne) [inline packages/manager/modules/support/src/modal/index.js:19]: In module you use url: '/betaHelpCenter', should to be same.

File: packages/manager/modules/support/src/modal/modal.html [hunk 6]
```diff
@@ -0,0 +1,29 @@
+<oui-modal
+    class="betaModal"
+    loading="$ctrl.isLoading"
```
Comment (ghyenne) [inline packages/manager/modules/support/src/modal/modal.html:3]: Isn't defined.

File: packages/manager/modules/support/src/modal/modal.routing.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  $stateProvider.state('support.tickets.beta-help-center', {
+    url: '/betaHelpCenter',
+    views: {
+      modal: {
+        component: 'modalBetaHelpcenter',
+      },
+    },
+    layout: 'modal',
+    resolve: {
+      features: /* @ngInject */ (ovhFeatureFlipping) => {
```
Comment (ghyenne) [inline packages/manager/modules/support/src/modal/modal.routing.js:14]: You can use a more explicit variable.

File: packages/manager/modules/support/src/modal/modal.routing.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        const features = FEATURES.map((feature) => `support:${feature}`);
+        return ovhFeatureFlipping.checkFeatureAvailability(features);
+      },
```
Comment (ghyenne) [inline packages/manager/modules/support/src/modal/modal.routing.js:17]:
```suggestion
      features: /* @ngInject */ (ovhFeatureFlipping) => ovhFeatureFlipping.checkFeatureAvailability(FEATURES),
```

**Review by ghyenne — CHANGES_REQUESTED** [review]

**Comment by stif59100** [comment]
I cleaned up my code.

It was not necessary to do a featureFlipping because the dispatch is done directly in the sidebar.

File: packages/manager/modules/support/src/modal/modal.scss [hunk 9]
```diff
@@ -0,0 +1,21 @@
+@import '@ovh-ux/ui-kit/dist/scss/tokens/_colors';
+@import '@ovh-ux/ui-kit/dist/scss/tokens/_fonts';
+@import '@ovh-ux/ui-kit/dist/scss/tokens/_globals';
+
+.betaModal {
+  background-image: url(../assets/images/background.png);
+}
+h1 {
```
Comment (cbourgois) [inline packages/manager/modules/support/src/modal/modal.scss:8]: IMO you shouldn't override global styles.

**Review by vks-ovh — APPROVED** [review]

File: packages/manager/modules/support/src/modal/translations/Messages_fr_FR.json [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+  "modal_beta_help_center_description": "Nouvelle interface de vos demandes d'assistance",
+  "modal_beta_help_center_please_note": "Découvrez la dès maintenant en basculant sur la nouvelle interface et faites-nous part de votre avis.",
```
Comment (mohammed-zahaf) [inline packages/manager/modules/support/src/modal/translations/Messages_fr_FR.json:3]: i am not stotaly sure but may be it's like this :
```suggestion
  "modal_beta_help_center_please_note": "Découvrez là dès maintenant en basculant sur la nouvelle interface et faites-nous part de votre avis.",
```

File: packages/manager/modules/support/src/modal/modal.html [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+                variant="link"
+                variant-nav="next"
+            >
+                <span data-translate="modal_beta_help_center_later"></span>
+            </oui-button>
+        </div>
+        <div class="col-6">
+            <img
+                data-ng-src="{{ $ctrl.illustration }}"
+                alt="Beta image modal"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/support/src/modal/modal.html:25]: I think it's a good practice to have put this.
So, we can add a French alternative description.

`alt="{{ 'ma_description' | translate }}"`

File: packages/manager/modules/support/src/modal/modal.constants.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
+const helpRoot = 'https://help-beta.ovhcloud.com/csm';
+const homeIndex = '-home?id=csm_index';
+
+export const GUIDES_LINK = {
+  DE: `${helpRoot}/de${homeIndex}`,
```
Comment (mohammed-zahaf) [inline packages/manager/modules/support/src/modal/modal.constants.js:5]: it's a good practice to have a default entry (en for example).
```suggestion
  DEFAULT: `${helpRoot}/en${homeIndex}`,
  DE: `${helpRoot}/de${homeIndex}`,
```

File: packages/manager/modules/support/src/modal/modal.controller.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import { GUIDES_LINK } from './modal.constants';
+
+export default class HelpCenterModalController {
+  /* @ngInject */
+  constructor($translate, coreConfig, $window) {
+    this.$translate = $translate;
+    this.coreConfig = coreConfig;
+    this.window = $window;
```
Comment (anooparveti) [inline packages/manager/modules/support/src/modal/modal.controller.js:8]:
```suggestion
    this.$window = $window;
```

File: packages/manager/modules/support/src/modal/modal.scss [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+.beta-modal {
+  background-image: url(../assets/images/background.png);
+  h1 {
+    font-size: 1.1rem;
+    color: $p-500;
+    font-weight: 600;
+  }
+  p {
+    color: $p-500;
```
Comment (anooparveti) [inline packages/manager/modules/support/src/modal/modal.scss:13]: **nitpick:** Can we directly use `oui-color-p-500` class on the required tag?

File: packages/manager/modules/support/src/modal/modal.scss [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+  }
+  img {
+    float: right;
```
Comment (anooparveti) [inline packages/manager/modules/support/src/modal/modal.scss:16]: **nitpick:** Use `float-right` from bootstrap

**Review by mohammed-zahaf — DISMISSED** [review]

File: packages/manager/modules/support/src/modal/modal.html [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+                class="oui-color-p-500"
+            ></p>
+            <oui-button ng-click="$ctrl.onClick()" variant="primary">
+                <span data-translate="modal_beta_help_center_access"></span
+            ></oui-button>
+
+            <oui-button
+                on-click="$ctrl.goBack()"
+                variant="link"
+                variant-nav="next"
```
Comment (anooparveti) [inline packages/manager/modules/support/src/modal/modal.html:23]:
```suggestion
                data-on-click="$ctrl.goBack()"
                data-variant="link"
                data-variant-nav="next"
```

**Review by anooparveti — APPROVED** [review]

File: packages/manager/modules/support/src/modal/translations/Messages_fr_FR.json [hunk 17]
```diff
@@ -0,0 +1,7 @@
+{
+  "modal_beta_help_center_description": "Nouvelle interface de vos demandes d'assistance",
+  "modal_beta_help_center_please_note": "Découvrez-là dès maintenant en basculant sur la nouvelle interface et faites-nous part de votre avis.",
+  "modal_beta_help_center_access": "Basculer vers la nouvelle interface",
+  "modal_beta_help_center_later": "Peut-être plus tard",
+  "modal_description_image": " Aperçu du centre d'aide"
```
Comment (ghyenne) [inline packages/manager/modules/support/src/modal/translations/Messages_fr_FR.json:6]:
```suggestion
  "modal_description_image": "Aperçu du centre d'aide"
```

**Review by ghyenne — CHANGES_REQUESTED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by ghyenne — DISMISSED** [review]

**Review by vks-ovh — APPROVED** [review]

**Comment by mohammed-zahaf** [comment]
Hello @stif59100 , i guess you have to target the `release/sprint-2242-2244` ?

File: packages/manager/modules/support/src/modal/modal.controller.js [hunk 18]
```diff
@@ -0,0 +1,18 @@
+import { GUIDES_LINK } from './modal.constants';
+
+export default class HelpCenterModalController {
+  /* @ngInject */
+  constructor($translate, coreConfig, $window) {
+    this.$translate = $translate;
```
Comment (antleblanc) [inline packages/manager/modules/support/src/modal/modal.controller.js:6]: **nitpick:** Looks like `$translate` is unused.

**Review by mohammed-zahaf — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**